pandas/core/indexes/base.py

```diff
@@ -1053,6 +1053,10 @@ def _to_embed(self, keep_tz=False, dtype=None):
 
     @Appender(_index_shared_docs['astype'])
     def astype(self, dtype, copy=True):
+        if is_categorical_dtype(dtype):
+            from .category import CategoricalIndex
```

Comment:
I guess we have an import issue if we import this at the top (with the fully qualified path)?